File: travelsAndTour-FrontEnd/src/app/servicess/booking.service.ts
import { Injectable } from '@angular/core';
import { Booking } from '../models/booking';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class BookingService {
serverurl="http://localhost:8080/tour"
  
  constructor(private http:HttpClient) { }
  getAll():Observable<Booking[]>
  {
    return this.http.get<Booking[]>(this.serverurl+"/booking")
  }
  insert(b:Booking):Observable<Booking>
  {
    return this.http.post<Booking>(this.serverurl+"/booking",b)
  }
  delete(id:any):Observable<any>
  {
    return this.http.delete(this.serverurl+"/booking/"+id)
  }
update(id:any,b:Booking):Observable<Booking>
{
  return this.http.put<Booking>(this.serverurl+"/booking/"+id,b)
}
search(id:any):Observable<Booking>
  {
    return this.http.get<Booking>(this.serverurl+"/booking/"+id)
  }
}

File: travel_and-tour_website-Backend/src/main/java/com/omkar/travel_and/tour_website/Backend/services/BookingService.java
package com.omkar.travel_and.tour_website.Backend.services;

import java.util.List;

import com.omkar.travel_and.tour_website.Backend.model.Booking;

public interface BookingService {
	Booking insert(Booking bo);
	void delete(long bid);
	Booking update(long bid,Booking bo);
	Booking search(long bid);
	List<Booking> getAll();
}
